research/RRDTool/src/rrdtool-1.0.0/doc/RRDs.3
.rn '' }`
''' $RCSfile$$Revision$$Date$
'''
''' $Log$
'''
.de Sh
.br
.if t .Sp
.ne 5
.PP
\fB\\$1\fR
.PP
..
.de Sp
.if t .sp .5v
.if n .sp
..
.de Ip
.br
.ie \\n(.$>=3 .ne \\$3
.el .ne 3
.IP "\\$1" \\$2
..
.de Vb
.ft CW
.nf
.ne \\$1
..
.de Ve
.ft R

.fi
..
'''
'''
'''     Set up \*(-- to give an unbreakable dash;
'''     string Tr holds user defined translation string.
'''     Bell System Logo is used as a dummy character.
'''
.tr \(*W-|\(bv\*(Tr
.ie n \{\
.ds -- \(*W-
.ds PI pi
.if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\" diablo 12 pitch
.ds L" ""
.ds R" ""
'''   \*(M", \*(S", \*(N" and \*(T" are the equivalent of
'''   \*(L" and \*(R", except that they are used on ".xx" lines,
'''   such as .IP and .SH, which do another additional levels of
'''   double-quote interpretation
.ds M" """
.ds S" """
.ds N" """""
.ds T" """""
.ds L' '
.ds R' '
.ds M' '
.ds S' '
.ds N' '
.ds T' '
'br\}
.el\{\
.ds -- \(em\|
.tr \*(Tr
.ds L" ``
.ds R" ''
.ds M" ``
.ds S" ''
.ds N" ``
.ds T" ''
.ds L' `
.ds R' '
.ds M' `
.ds S' '
.ds N' `
.ds T' '
.ds PI \(*p
'br\}
.\"	If the F register is turned on, we'll generate
.\"	index entries out stderr for the following things:
.\"		TH	Title 
.\"		SH	Header
.\"		Sh	Subsection 
.\"		Ip	Item
.\"		X<>	Xref  (embedded
.\"	Of course, you have to process the output yourself
.\"	in some meaninful fashion.
.if \nF \{
.de IX
.tm Index:\\$1\t\\n%\t"\\$2"
..
.nr % 0
.rr F
.\}
.TH RRDs 3pm "rrdtool-1.0.0" "15/Jul/99" "rrdtool"
.UC
.if n .hy 0
.if n .na
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.de CQ          \" put $1 in typewriter font
.ft CW
'if n "\c
'if t \\&\\$1\c
'if n \\&\\$1\c
'if n \&"
\\&\\$2 \\$3 \\$4 \\$5 \\$6 \\$7
'.ft R
..
.\" @(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2
.	\" AM - accent mark definitions
.bd B 3
.	\" fudge factors for nroff and troff
.if n \{\
.	ds #H 0
.	ds #V .8m
.	ds #F .3m
.	ds #[ \f1
.	ds #] \fP
.\}
.if t \{\
.	ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.	ds #V .6m
.	ds #F 0
.	ds #[ \&
.	ds #] \&
.\}
.	\" simple accents for nroff and troff
.if n \{\
.	ds ' \&
.	ds ` \&
.	ds ^ \&
.	ds , \&
.	ds ~ ~
.	ds ? ?
.	ds ! !
.	ds /
.	ds q
.\}
.if t \{\
.	ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.	ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.	ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.	ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.	ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.	ds ? \s-2c\h'-\w'c'u*7/10'\u\h'\*(#H'\zi\d\s+2\h'\w'c'u*8/10'
.	ds ! \s-2\(or\s+2\h'-\w'\(or'u'\v'-.8m'.\v'.8m'
.	ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.	ds q o\h'-\w'o'u*8/10'\s-4\v'.4m'\z\(*i\v'-.4m'\s+4\h'\w'o'u*8/10'
.\}
.	\" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds v \\k:\h'-(\\n(.wu*9/10-\*(#H)'\v'-\*(#V'\*(#[\s-4v\s0\v'\*(#V'\h'|\\n:u'\*(#]
.ds _ \\k:\h'-(\\n(.wu*9/10-\*(#H+(\*(#F*2/3))'\v'-.4m'\z\(hy\v'.4m'\h'|\\n:u'
.ds . \\k:\h'-(\\n(.wu*8/10)'\v'\*(#V*4/10'\z.\v'-\*(#V*4/10'\h'|\\n:u'
.ds 3 \*(#[\v'.2m'\s-2\&3\s0\v'-.2m'\*(#]
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.ds oe o\h'-(\w'o'u*4/10)'e
.ds Oe O\h'-(\w'O'u*4/10)'E
.	\" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.	\" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.	ds : e
.	ds 8 ss
.	ds v \h'-1'\o'\(aa\(ga'
.	ds _ \h'-1'^
.	ds . \h'-1'.
.	ds 3 3
.	ds o a
.	ds d- d\h'-1'\(ga
.	ds D- D\h'-1'\(hy
.	ds th \o'bp'
.	ds Th \o'LP'
.	ds ae ae
.	ds Ae AE
.	ds oe oe
.	ds Oe OE
.\}
.rm #[ #] #H #V #F C
.SH "NAME"
RRDs \- Access rrdtool as a shared module
.SH "SYNOPSIS"
.PP
.Vb 8
\&  use RRDs::ext
\&  RRDs::error
\&  RRDs::last ...
\&  RRDs::create ...
\&  RRDs::update ...
\&  RRDs::graph ...
\&  RRDs::fetch ...
\&  RRDs::tune ...
.Ve
.SH "DESCRIPTION"
.Sh "Calling Sequence"
This module accesses rrdtool functionality directly from within perl. The
arguments to the functions listed in the \s-1SYNOPSIS\s0 are explained in the regular
rrdtool documentation. The commandline call
.PP
.Vb 1
\& rrdtool update mydemo.rrd N:12:13
.Ve
gets turned into
.PP
.Vb 1
\& RRDs::update qw(mydemo.rrd N:12:13);
.Ve
.Sh "Error Handling"
The \s-1RRD\s0 functions will not abort your program even when they can not make
sense out of the arguments you fed them. There are two ways to determine if
an error has occured.
.PP
First the every function will return the value \-1 if an error occured.
Second, the function RRDs::error can be called to get the error message
from the last function call. If RRDs::error does not return an error
then the previous function has completed its task succesfully.
.PP
.Vb 3
\& RRDs::update qw(mydemo.rrd N:12:13);
\& my $ERR=RRDs::error;
\& die "ERROR while updating mydemo.rrd: $ERR\en" if $ERR;
.Ve
.Sh "Return Values"
The functions RRDs::last, RRDs::graph and RRDs::fetchgive return their
findigs.
.PP
RRDs::last returns a single \s-1INTEGER\s0 representing the last update time.
.PP
.Vb 1
\& $lastupdate = RRDs::last ...
.Ve
RRDs::graph returns an pointer to an \s-1ARRAY\s0 containing the x-size and y-size of the
created gif and results of the \s-1PRINT\s0 arguments.
.PP
.Vb 3
\& ($averages,$xsize,$ysize) = RRDs::graph ...
\& print "Gifsize: ${xsize}x${ysize}\en";
\& print "Averages: ", (join ", ", @$averages);
.Ve
RRDs::fetch is the most complex of the pack regarding return values. There are
4 values. Two normal integers, a pointer to an array and a pointer to a
array of pointers.
.PP
.Vb 13
\& my ($start,$step,$names,$data) = RRDs::fetch ... 
\& print "Start:        ".localtime($start)."\en"; 
\& print "Stepsize:     $step seconds\en"; 
\& print "Column Names: ".join (", ", @$names)."\en";
\& print "Date:\en";
\& foreach my $line (@$array){
\&      print "".localtime($start),"   ";
\&      $start += $step; 
\&      foreach my $val (@$line) {                
\&           printf "%12.1f", $val;
\&      }
\&      print "\en";
\& }
.Ve
See the examples directory for more ways to use this extension.
.SH "AUTHOR"
Tobias Oetiker <oeitker@ee.ethy.ch>

.rn }` ''
.IX Title "RRDs 3pm"
.IX Name "RRDs - Access rrdtool as a shared module"

.IX Header "NAME"

.IX Header "SYNOPSIS"

.IX Header "DESCRIPTION"

.IX Subsection "Calling Sequence"

.IX Subsection "Error Handling"

.IX Subsection "Return Values"

.IX Header "AUTHOR"



research/RRDTool/src/rrdtool-1.0.0/perl-shared/RRDs.pm
package RRDs;

use strict;
use vars qw(@ISA $VERSION);

@ISA = qw(DynaLoader);

require DynaLoader;

$VERSION = 1.000001                                 ;

bootstrap RRDs $VERSION;

1;
__END__

=head1 NAME

RRDs - Access rrdtool as a shared module

=head1 SYNOPSIS

  use RRDs::ext
  RRDs::error
  RRDs::last ...
  RRDs::create ...
  RRDs::update ...
  RRDs::graph ...
  RRDs::fetch ...
  RRDs::tune ...

=head1 DESCRIPTION

=head2 Calling Sequence

This module accesses rrdtool functionality directly from within perl. The
arguments to the functions listed in the SYNOPSIS are explained in the regular
rrdtool documentation. The commandline call

 rrdtool update mydemo.rrd N:12:13

gets turned into

 RRDs::update qw(mydemo.rrd N:12:13);

=head2 Error Handling

The RRD functions will not abort your program even when they can not make
sense out of the arguments you fed them. There are two ways to determine if
an error has occured.

First the every function will return the value -1 if an error occured.
Second, the function RRDs::error can be called to get the error message
from the last function call. If RRDs::error does not return an error
then the previous function has completed its task succesfully.

 RRDs::update qw(mydemo.rrd N:12:13);
 my $ERR=RRDs::error;
 die "ERROR while updating mydemo.rrd: $ERR\n" if $ERR;

=head2 Return Values

The functions RRDs::last, RRDs::graph and RRDs::fetchgive return their
findigs.

RRDs::last returns a single INTEGER representing the last update time.

 $lastupdate = RRDs::last ...

RRDs::graph returns an pointer to an ARRAY containing the x-size and y-size of the
created gif and results of the PRINT arguments.

 ($averages,$xsize,$ysize) = RRDs::graph ...
 print "Gifsize: ${xsize}x${ysize}\n";
 print "Averages: ", (join ", ", @$averages);

RRDs::fetch is the most complex of the pack regarding return values. There are
4 values. Two normal integers, a pointer to an array and a pointer to a
array of pointers.

 my ($start,$step,$names,$data) = RRDs::fetch ... 
 print "Start:        ".localtime($start)."\n"; 
 print "Stepsize:     $step seconds\n"; 
 print "Column Names: ".join (", ", @$names)."\n";
 print "Date:\n";
 foreach my $line (@$array){
      print "".localtime($start),"   ";
      $start += $step; 
      foreach my $val (@$line) {		
           printf "%12.1f", $val;
      }
      print "\n";
 }

See the examples directory for more ways to use this extension.

=head1 AUTHOR

Tobias Oetiker <oeitker@ee.ethy.ch>

=cut
